docs/LLS_Metallicities-Wotta/cloudyiongrid.py

```diff
@@ -0,0 +1,932 @@
+        redshift   -> the redshift
+        density    -> the total hydrogen volume density [log]
+        fstar      -> the strength of the dust deplition [linear]
+
+        """
+
+        #first init the minimal model 
+        self.minimal(column,metal,redshift,density)
+
+        #update as needed global variable
+        self.fstar=fstar
+        self.root="n{0}_r{1}_c{2}_Z{3}_f{4}".format(self.density,self.redshift,self.column,self.metal,self.fstar)
+
+        #now compute the dust scaling
+        if(self.fstar > 0):
+            #If fstar > 0 ,add dust
+            self.dustscale=self.fstar*(10**self.metal)
+        else:
+            #Else, do not include dust
+            self.dustscale=False
+
+        #now compute the abundances depletion following the method by Jenkins et al. 
+        #Going to f = -1 brings the corrections close to 0
+        ejelm=['Carbon','Nitrogen','Oxygen','Magnesium','Silicon','Phosphorus','Chlorine','Titanium',\
+                        'Chromium','Manganese','Iron','Nickel','Copper','Zinc']
+        ejAx=[-0.101,-0.000,-0.225,-0.997,-1.136,-0.945,-1.242,-2.048,-1.447,-0.857,-1.285,-1.490,-0.710,-0.610]
+        ejBx=[-0.193,-0.109,-0.145,-0.800,-0.570,-0.166,-0.314,-1.957,-1.508,-1.354,-1.513,-1.829,-1.102,-0.279]
+        ejzx=[0.803,0.550,0.598,0.531,0.305,0.488,0.609,0.430,0.470,0.520,0.437,0.599,0.711,0.555]
+
+        #init depletion
+        self.abundance={}
+
+        #compute values
+        for ee in range(len(ejelm)):
+            depl=ejBx[ee]+ejAx[ee]*(self.fstar-ejzx[ee])
+            if(depl < 0):
+                self.abundance[ejelm[ee]]=10**depl
+            else:
+                self.abundance[ejelm[ee]]=1.0
+
+
+    def modsource(self,column,metal,redshift,density,jgal,jqso):
+
+
+        """
+        Specify a cloudy model with varying SED properties 
+
+        column     -> the HI column density [log]
+        metal      -> the log metallicity in solar unit [X/H] 
+        redshift   -> the redshift
+        density    -> the total hydrogen volume density [log]
+        jgal       -> weight to add in the galaxy contribution
+        jqso       -> weight to add in the qso contribution
+
+        """
+
+        #import numpy as np
+        #from astropy import constants as const
+        #from scipy import interpolate
+        #import mypython as mp
+
+        #first init the minimal model 
+        self.minimal(column,metal,redshift,density)
+
+        #make copy of uvb
+        uvb_nu=self.source['lgnu']
+        uvb_f=self.source['lgfpiJ']
+
+        #next update coefficient and name
+        self.jgal=jgal
+        self.jqso=jqso
+        self.root="n{0}_r{1}_c{2}_Z{3}_g{4}_q{5}".format(self.density,self.redshift,self.column,self.metal,self.jgal,self.jqso)
+
+        ##################
+        ###  quasar    ###
+        ##################
+
+        #now load qso sed
+        qso_nu=[]
+        qso_f=[]
+
+        fil=open('Scripts/richards_qsosed.txt')
```

Comment:
This was Michele's.  I'm not sure where that file is, so I can't put it anywhere.